modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -214,7 +207,11 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
             if (kernel_size.empty() || kernel_size.size() == 2)
                 return backendId == DNN_BACKEND_OPENCV ||
                        (backendId == DNN_BACKEND_HALIDE && haveHalide() &&
-                           (type == MAX || (type == AVE && !pad_t && !pad_l && !pad_b && !pad_r)));
+                       (type == MAX || (type == AVE && !pads_begin[0] && !pads_begin[1] && !pads_end[0] && !pads_end[1])));
+            if (kernel_size.size() == 1)
+                return (backendId == DNN_BACKEND_OPENCV && preferableTarget == DNN_TARGET_CPU) ||
+                       (backendId == DNN_BACKEND_OPENCV && preferableTarget == DNN_TARGET_OPENCL) ||
+                       (backendId == DNN_BACKEND_OPENCV && preferableTarget == DNN_TARGET_OPENCL_FP16);
```

Comment:
> supportBackend(int backendId)

is not supposed to check `preferableTarget` value at all.

---

It is a design bug.
Caller doesn't expect failures for `DNN_BACKEND_OPENCV`.

/cc @dkurt 

---

`return backendId == DNN_BACKEND_OPENCV;` should be enough here (replace these 3 lines).

Reply:
Actually, there is already `else if (backendId == DNN_BACKEND_OPENCV || backendId == DNN_BACKEND_HALIDE)` so this condition is always true

Reply:
It is `false` if `backendId == DNN_BACKEND_HALIDE`.